README.md

```diff
@@ -1 +1,238 @@
+
+You can one of the following patterns:
+
+1. `module.exports= {...}`
+1. `module.exports=function() {...}`
+1. `exports.method = function() {...}`
+1. `exports.object = {...}`
+
+2 and 3 can also be use for classes (factory function or functional inheritance pattern). We won't discuss classes in great detail since it's more of general JavaScript topic and not exclusive to Node. 
+
+Right now let's cover the difference between assigning an object and a method.
+
+## Exporting Objects vs. Methods
+
+Consider an example where we have a configuration file with some settings (`configs.js`):
+
+```js
+module.exports = {
+  url: 'http://webapplog.com',
+  name: 'React Quicly',
+  port: 3000,
+  apiKey: "D1FCFE58-575F-40FF-AF36-0CBD6A367756"
+}
+```
+
+In the `main.js` file (which uses the module), we can import the module and use any settings by accessing the property by the name such as `url` or `name` (`.js` extension is optional):
+
+```js
+var configs = require('./configs')
+console.log('URL is %s', configs.url)
+```
+
+The result of running `node main` will be the printed URL with the value supplied by the `configs.js`:
+
+```
+URL is http://webapplog.com
+```
+
+We achieved modularity! Now we can have many other files like `program.js` or `app.js` which can import the `config.js` file and access the same values. If we ever need to change `port` or `apiKey`, then it's effortless. We just update `configs.js`. No need to search in all files and risk bugs by missing a few.
+
+However, there's a limitation to this approach. Let's say we need to have three sets of API keys: one for development, one for testing and one for production. You can write `if/else` conditions in `main.js` but then you end up repeating the same logic over and over in other files which need to use the configurations. 
+This approach isn't great because we end up with a lot of logic which really belongs to the configuration module spread out across various main files (files which import the module). Take a look at the code below (also in `main-envirionment-bad.js`):
+
+```js
+if (process.env.NODE_ENV == 'production') {
+  var configs = require('./configs-production')
+} else if (process.env.NODE_ENV == 'testing) {
+  var configs = require('./configs-testing')
+} else { // Assume development
+  var configs = require('./configs')
+}
+
+console.log('URL is %s', configs.url)
+```
+
+If you are thinking we should move the logic to the `config.js` file, the you are right. But how do we go about it if it's just an object? We need to use the `module.exports = function() {}` pattern which is more versatile. It allows us to execute some logic and even use parameters from the function's arguments. 
```

Comment:
end after the `But how do we go about it if it's just an object?`

and create a new header to start next section about next pattern